woocommerce/woocommerce-android
# [UI Tests] Add support for real API UI tests.

Note: this PR was replaced with identical #8689, [see why](https://github.com/woocommerce/woocommerce-android/pull/8636#issuecomment-1490384315).

### What + Why
This PR adds a possibility to selectively run UI tests either against a mocked API (WireMock) or the real API. It also adds a sample test as a proof of concept.

Using each kind of API had its pros and cons, this change allows making a choice and experiment with having some of our UI tests hit the real API.

### How
I went for a somewhat primitive solution (TBH, it's also the _only_ easy solution I found so far) and aimed for the minimal footprint. 
- There exists a [MockingInterceptor](https://github.com/woocommerce/woocommerce-android/blob/trunk/WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/helpers/MockingInterceptor.kt) class of `Interceptor` type, it redirects all HTTP requests for `public-api.wordpress.com` to a local port associated with WireMock server. I added a global `useMockedAPI` boolean variable which defines if the redirect should happen for a given test. If it does not happen, real API is used. All the infrastructure related to WireMock still runs in all cases.
- Added a `e2eRealAPIAssertStoreTitle` test that logs into a real store and checks the store title.
- The credentials for real store are added to mobile-secrets (not a part of PR)
- Real store is tied to a WPCOM (non-a8c) account.

### Testing instructions
1. You can run all `e2e...` tests locally (make sure you're using latest mobile secrets, and then run `./gradlew configureApply` to obtain test store credentials from secrets) and that old tests still use mocks, whereas the new test uses a real API.
2. The tests also pass on [FTL](https://console.firebase.google.com/u/1/project/api-project-108380595987/testlab/histories/bh.edfd947f2636efe3/matrices/5079266829039065525/details?stepId=bs.7db9e66f0e276a20&testCaseId=44).
